packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1298,38 +1295,21 @@ function getRealPath(resolved: string, preserveSymlinks?: boolean): string {
 }
 
 /**
- * if importer was not resolved by vite's resolver previously
- * (when esbuild resolved it)
- * resolve importer's pkg and add to idToPkgMap
+ * Load closest `package.json` to `importer`
  */
 function resolvePkg(importer: string, options: InternalResolveOptions) {
```

Comment:
I had to refactor this function, since it's incorrectly using `resolvePackageData` and looks unnecessarily complex 🤔 What happens is that this function gets the nearest `package.json`, and loads the `PackageData`. And the new changes does the same too.

In the old changes, it manually walks `/User/foo/project`, create a `possiblePkgIds` of `['/User/foo/project', '/User/foo', '/User']`, and then try to resolve the `package.json` in those directories. So `resolvePackageData` would receive absolute paths instead of bare imports which doesn't work with this refactor.